src/Datadog.Trace/ExtensionMethods/HeadersCollectionExtensions.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Globalization;
+using Datadog.Trace.Headers;
+
+namespace Datadog.Trace.ExtensionMethods
+{
+    /// <summary>
+    /// Extension methods for <see cref="IHeadersCollection"/>.
+    /// </summary>
+    public static class HeadersCollectionExtensions
+    {
+        private const NumberStyles NumberStyles = System.Globalization.NumberStyles.Integer;
+        private static readonly CultureInfo InvariantCulture = CultureInfo.InvariantCulture;
+
+        /// <summary>
+        /// Creates a <see cref="SpanContext"/> from the values found in the specified headers.
+        /// </summary>
+        /// <param name="headers">The headers that contain the values to be extracted.</param>
+        /// <returns>A new <see cref="SpanContext"/> that contains values extracted from <paramref name="headers"/>.</returns>
+        public static SpanContext ExtractSpanContext(this IHeadersCollection headers)
+        {
+            if (headers == null)
+            {
+                throw new ArgumentNullException(nameof(headers));
```

Comment:
You could just return `null` here instead of throwing an exception. Maybe just move a log line into here, could save a try/catch above.

Reply:
Be good to know if parsing fails, but successfully parsing isn't a requirement.

Reply:
This is an extension method so you call it using the same syntax as an instance method:

```csharp
headers.InjectSpanContext(...);
```

To keep the code idiomatic, I think extension methods should behave like instance methods and throw an exception if they are called on a null reference.

However, to your concern, I made sure that our code paths never call this with null headers (161eda0a9ddbc9c5dd05cae8d91baf600fe39bac) and added logging if a header is present we we are unable to parse it (13c65ec46c8636bd434013f28cce002aef60cc2e).

Reply:
Cool, totally fine. Whatever makes the most sense for dotnet